src/components/TextField/TextField.tsx

```diff
@@ -212,7 +212,7 @@ class TextField extends React.PureComponent<CombinedProps, State> {
       </div>
     ) : null;
 
-    const characterCount = value.length;
+    const characterCount = value ? value.length : 0;
```

Comment:
```suggestion
    const characterCount = typeof value === 'string' ? value.length : 0;
```

Let be explicit in checking that value is indeed a string.